staging/src/k8s.io/api/storage/v1beta1/types.go

```diff
@@ -209,3 +209,174 @@ type VolumeError struct {
 	// +optional
 	Message string `json:"message,omitempty" protobuf:"bytes,2,opt,name=message"`
 }
+
+// VolumePluginMechanism is the type of mechanism components should use for volume operations
+type VolumePluginMechanism string
+
+const (
+	// VolumePluginMechanismInTree means components should use the in-tree plugin for volume operations
+	VolumePluginMechanismInTree VolumePluginMechanism = "in-tree"
+	// VolumePluginMechanismCSI means components should use the CSI Driver for volume operations
+	VolumePluginMechanismCSI VolumePluginMechanism = "csi"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CSIDriver captures information about a Container Storage Interface (CSI)
+// volume driver deployed on the cluster.
+// CSIDriver objects are non-namespaced.
+type CSIDriver struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Standard object metadata.
+	// metadata.Name indicates the name of the CSI driver that this object
```

Comment:
Can you repeat the validation rules here, so I don't have to go digging to understand it?

Reply:
Sure.  The driver name must be 63 characters or less, beginning and ending with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), dots (.), and alphanumerics between.